python/mdvtools/dbutils/mdv_server_app.py

```
@@ -710,6 +723,78 @@
+                        extracted_list = os.listdir(temp_extract_path)
+
+                        extracted_project_path = None
+
+                        # Check if the directory or sub-directory is valid mdv project
+                        if is_valid_mdv_project(temp_extract_path):
+                            extracted_project_path = temp_extract_path
+                        else:
+                            for e in extracted_list:
+                                sub_path = os.path.join(temp_extract_path, e)
+                                if os.path.isdir(sub_path):
+                                    if is_valid_mdv_project(sub_path):
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)